test/integration/examples/apiserver_test.go

```diff
@@ -184,24 +186,38 @@ func TestAggregatedAPIServer(t *testing.T) {
 			t.Fatal(err)
 		}
 		atomic.StoreInt32(wardlePort, int32(port))
-		options := sampleserver.NewWardleServerOptions(os.Stdout, os.Stderr)
-		options.Complete()
-
-		options.RecommendedOptions.SecureServing.BindAddress = net.ParseIP("127.0.0.1")
-		options.RecommendedOptions.SecureServing.BindPort = port
-		options.RecommendedOptions.SecureServing.Listener = listener
-		options.RecommendedOptions.Etcd.StorageConfig.ServerList = []string{framework.GetEtcdURL()}
-		options.RecommendedOptions.SecureServing.ServerCert.CertDirectory = wardleCertDir
-		options.RecommendedOptions.Authentication.RequestHeader.UsernameHeaders = []string{"X-Remote-User"}
-		options.RecommendedOptions.Authentication.RequestHeader.GroupHeaders = []string{"X-Remote-Group"}
-		options.RecommendedOptions.Authentication.RequestHeader.ExtraHeaderPrefixes = []string{"X-Remote-Extra-"}
-		options.RecommendedOptions.Authentication.RequestHeader.ClientCAFile = proxyCACertFile.Name()
-		options.RecommendedOptions.Authentication.RequestHeader.AllowedNames = []string{"kube-aggregator"}
-		options.RecommendedOptions.Authentication.RemoteKubeConfigFile = kubeconfigFile.Name()
-		options.RecommendedOptions.Authorization.RemoteKubeConfigFile = kubeconfigFile.Name()
-		options.RecommendedOptions.CoreAPI.CoreAPIKubeconfigPath = kubeconfigFile.Name()
-
-		if err := options.RunWardleServer(stopCh); err != nil {
+
+		wardleCmd := sampleserver.NewCommandStartWardleServer(os.Stdout, os.Stderr, stopCh)
+		o := sampleserver.NewWardleServerOptions(os.Stdout, os.Stderr)
+		o.RecommendedOptions.SecureServing.Listener = listener
+		// parse flags
+		flags := wardleCmd.Flags()
+		o.RecommendedOptions.AddFlags(flags)
```

Comment:
would it help to pass the initial `WardleServerOptions` (with a set listener) to `NewCommandStartWardleServer`? could avoid this acrobatic with flags and RunE.

Reply:
i can not find a way to pass to the initial `WardleServerOptions `. Because the `wardleCmd ` does not expose it.

Reply:
Or we have another choice modify `NewCommandStartWardleServer` return the options.

Reply:
Why not pre-create it and inject into `NewCommandStartWardleServer`. Is the signature of `NewCommandStartWardleServer` fixed?

Reply:
We can't, because the `NewCommandStartWardleServer` does not take it as an argument.